src/App.jsx

```diff
@@ -14,33 +17,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function prepareGoods(sortType, reverse) {
+  const goods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  goods.sort((good1, good2) => {
+    switch (sortType) {
+      case SORT_BY_ALPHABET:
+        return good1.localeCompare(good2);
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+      case SORT_BY_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reverse) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+
+export const App = () => {
+  const [sortType, setSortType] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = prepareGoods(sortType, isReversed);
+  const showResetButton = isReversed || sortType;
```

Comment:
```suggestion
  const isResetButtonVisible = isReversed || sortType;
```
Boolean variables should start from `is` or `has`